src/main/java/jewellery/inventory/service/ProductService.java

```diff
@@ -57,30 +60,47 @@ public ProductReturnResponseDto getProductReturnResponseDto(
 
   public List<ProductResponseDto> getAllProducts() {
     List<Product> products = productRepository.findAll();
+    logger.debug("Fetching all products");
     return products.stream().map(productMapper::mapToProductResponseDto).toList();
   }
 
   public List<ProductResponseDto> getByOwner(UUID ownerId) {
     List<Product> products = productRepository.findAllByOwnerId(ownerId);
+    logger.info("Get product by owner with ID: {}", ownerId);
     return products.stream().map(productMapper::mapToProductResponseDto).toList();
   }
 
   public Product getProduct(UUID id) {
+    logger.info("Get product by ID: {}", id);
     return productRepository.findById(id).orElseThrow(() -> new ProductNotFoundException(id));
   }
 
   public ProductResponseDto getProductResponse(UUID id) {
+    logger.info("Get productResponse by ID: {}", id);
     return productMapper.mapToProductResponseDto(getProduct(id));
   }
 
   public void updateProductOwnerAndSale(Product product, User newOwner, Sale sale) {
     updateProductOwnerRecursively(product, newOwner);
-    product.setPartOfSale(sale);
+    if (sale != null) {
+      product.setPartOfSale(sale);
+    } else {
+      product.setPartOfSale(null);
+    }
```

Comment:
This is the same as:
```suggestion
    product.setPartOfSale(sale);
```